backend/typescript/services/implementations/campService.ts

```diff
@@ -125,6 +160,18 @@ class CampService implements ICampService {
       throw error;
     }
 
+    if (waitlistedCamperId) {
+      camp.campSessions = (camp.campSessions as CampSession[]).filter(
+        (campSession) => campSession.id === campSessionId,
+      );
+
+      if (camp.campSessions.length === 0) {
+        throw new Error(
+          `No camp session matches with session ${campSessionId}`,
+        );
+      }
+    }
+
```

Comment:
can we also check the camp session's capacity here and throw an error if it's full? I think it's better than doing the check on the frontend, avoids doing a redirect? what do you think

Reply:
I think we agreed that the check would be done on the frontend, but yeah I can add that.